lib/deps/ajax.js

```diff
@@ -129,7 +129,7 @@ function ajax(options, adapterCallback) {
     if (response.statusCode >= 200 && response.statusCode < 300) {
       onSuccess(data, response, callback);
     } else {
-      if (options.binary) {
+      if (options.binary && data.length) {
```

Comment:
Oh, because this is an error, not entirely sure the `options.binary` check should be in there at all, it seems like

    try { 
      data = JSON.parse(data.toString());
    } catch(e) {}
    error = errors.generateErrorFromResponse(data);
    error.status = response.statusCode;
    callback(error);

Would what we should do?